roles/libvirt_manager/tasks/generate_networking_data.yml

```diff
@@ -0,0 +1,172 @@
+
+    - name: Generate VM list and associate MAC address
+      ansible.builtin.import_tasks: generate_macs.yml
+
+    - name: Dump MAC mapping
+      ansible.builtin.copy:
+        dest: "{{ _nic_info }}"
+        content: "{{ cifmw_libvirt_manager_mac_map | to_nice_yaml }}"
+# END MAC pre-generation management
+#
+# START generate all IPs using networking_mapper role/module
+- name: Inject all VMs in the inventory
+  vars:
+    _vm_type: "{{ item.key | regex_replace('(cifmw\\-)?([a-z]+)\\-[0-9]+', '\\2') }}"
+    _group: "{{ (_vm_type == 'crc') | ternary('ocp', _vm_type) }}"
+  ansible.builtin.add_host:
+    name: "{{ item.key }}"
+    groups: "{{ _group }}s"
+  loop: "{{ cifmw_libvirt_manager_mac_map | dict2items }}"
+
+- name: Generate all IPs based on MAC and networks
+  vars:
+    cifmw_networking_mapper_ifaces_info: >-
+      {{ cifmw_libvirt_manager_mac_map }}
+    cifmw_networking_mapper_network_name: "{{ cifmw_libvirt_manager_pub_net }}"
+    cifmw_networking_mapper_gather_facts: false
+  ansible.builtin.import_role:
+    name: networking_mapper
+
+- name: Manage networks if needed
+  when:
+    - _cifmw_libvirt_manager_layout.networks is defined
+  ansible.builtin.include_tasks: create_networks.yml
+
+- name: Craft patch for cifmw_networking_definition
+  vars:
+    _match: "^.*{{ cifmw_libvirt_manager_pub_net }}$"
+    _lnet_data: >-
+      {{
+        _network_data |
+        selectattr('name', 'match', _match) | first
+      }}
+    _networks: |
+      {{ _lnet_data.name | replace('cifmw_', '') }}:
+      {% if _lnet_data.ranges[0].start_v4 is defined and _lnet_data.ranges[0].start_v4 | length > 0                     %}
+        {% set net_4 = _lnet_data.ranges[0].start_v4 | ansible.utils.ipsubnet(_lnet_data.ranges[0].prefix_length_v4)    %}
+        network-v4: {{ net_4}}
+      {% endif                                                                                                          %}
+      {% if _lnet_data.ranges[0].start_v6 is defined and _lnet_data.ranges[0].start_v6 | length > 0                     %}
+        {% set net_6 = _lnet_data.ranges[0].start_v6 | ansible.utils.ipsubnet(_lnet_data.ranges[0].prefix_length_v6)  %}
+        network-v6: {{ net_6 }}
+      {% endif                                                                                                          %}
+    _gr_templates: |
+      {% for group in (_cifmw_libvirt_manager_layout.vms.keys() | sort) if group != 'controller' and
+                                                                           (_cifmw_libvirt_manager_layout.vms[group].amount | default(1) | int) > 0 %}
+      {% set _gr = (group == 'crc') | ternary('ocp', group)                                                             %}
+        {{ _gr }}s:
+          networks:
+            {{ _lnet_data.name | replace('cifmw_', '') }}:
+      {% if cifmw_networking_definition['group-templates'][_gr ~ 's']['network-template'] is not defined                %}
```

Comment:
@pablintino as I understand we have to segment the network in "some way". My idea here is to either benefit of any existing `network-template` set for the group, or fallback on the "segment" used by `internalapi`
But it really look bad depending on the format of that "segment", since it may be an actual range with the whole network hardcoded in it...

No better idea for now, maybe you have a trick?